plsync/mlabconfig.py

```diff
@@ -174,33 +189,64 @@ def export_experiment_records(output, sites, experiments):
+            export_experiment_records_v6(
+                output, sites, experiment, decoration='v6', flatnames=True)
+
+
+def export_experiment_records_v4(output,
+                                 sites,
+                                 experiment,
+                                 decoration='',
+                                 flatnames=False):
+    comment(output, '%s v4%s%s' % (experiment.dnsname(), (
+        ' decorated' if decoration else ''), (' flattened'
+                                              if flatnames else '')))
     for site in sites:
         # TODO: change site['nodes'] to a pre-sorted list type.
         for node in sorted(site['nodes'].values(), key=lambda n: n.hostname()):
             # TODO: remove sitenames (or exclude mlab4's).
-            write_a_record(output, experiment.sitename(node, decoration),
-                           experiment.ipv4(node))
-            write_a_record(output, experiment.recordname(node, decoration),
-                           experiment.ipv4(node))
+            sitename = experiment.sitename(node, decoration)
+            recordname = experiment.recordname(node, decoration)
+
+            if flatnames:
+                sitename = flatten_hostname(sitename)
+                recordname = flatten_hostname(recordname)
 
+            write_a_record(output, sitename, experiment.ipv4(node))
+            write_a_record(output, recordname, experiment.ipv4(node))
 
-def export_experiment_records_v6(output, sites, experiment, decoration=''):
-    comment(output, '%s v6%s' % (experiment.dnsname(), (' decorated' if
-                                                        decoration else '')))
+
+def export_experiment_records_v6(output,
+                                 sites,
+                                 experiment,
+                                 decoration='',
+                                 flatnames=False):
+    comment(output, '%s v6%s%s' % (experiment.dnsname(), (
+        ' decorated' if decoration else ''), (' flattened'
+                                              if flatnames else '')))
     for site in sites:
         # TODO: change site['nodes'] to a pre-sorted list type.
         for node in sorted(site['nodes'].values(), key=lambda n: n.hostname()):
             # TODO: remove sitenames (or exclude mlab4's).
             if (node.ipv6_is_enabled() and experiment.ipv6(node)):
-                write_aaaa_record(output, experiment.sitename(node, decoration),
-                                  experiment.ipv6(node))
-                write_aaaa_record(output, experiment.recordname(node,
-                                                                decoration),
-                                  experiment.ipv6(node))
+                sitename = experiment.sitename(node, decoration)
+                recordname = experiment.recordname(node, decoration)
+
+                if flatnames:
+                    sitename = flatten_hostname(sitename)
```

Comment:
Good point. I've made another commit to fix this, and also updated the unittest to not only check for the right things in the results, but to check that the things we don't want (i.e., sitename records) actually don't exist.  PTAL?